twill-core/src/main/java/org/apache/twill/internal/TwillContainerLauncher.java

```diff
@@ -210,21 +210,22 @@ protected void doStartUp() {
 
     @Override
     protected void doShutDown() {
-      // Wait for sometime for the container to stop
-      // TODO: Use configurable value for stop time (TWILL-192)
-      int maxWaitSecs = Constants.APPLICATION_MAX_STOP_SECONDS - 30;
-      maxWaitSecs = maxWaitSecs < 15 ? 15 : maxWaitSecs;
+      // Wait for sometime for the container to stop. Subtracts 15 seconds from the timeout to accommodate for
+      // the message propagation delay.
+      long timeoutMillis = getTerminationTimeoutMillis(Constants.APPLICATION_MAX_STOP_SECONDS, TimeUnit.SECONDS);
+      timeoutMillis = timeoutMillis < 30L ? timeoutMillis : timeoutMillis - 15L;
```

Comment:
The unit of the timeout here changes from secs -> millis, should we minus 15000L here?

Reply:
Actually the timeout should just be in seconds as the rest of the logic are all in seconds.